contracts/rewards/src/msg.rs

```diff
@@ -60,4 +61,15 @@ pub enum ExecuteMsg {
 
 #[cw_serde]
 #[derive(QueryResponses)]
-pub enum QueryMsg {}
+pub enum QueryMsg {
+    /// Gets the rewards pool details for the given `pool_id``
+    #[returns(RewardsPoolResponse)]
+    GetRewardsPool { pool_id: PoolId },
+}
+
+#[cw_serde]
+pub struct RewardsPoolResponse {
+    pub balance: Uint128,
+    pub epoch_duration: Uint64,
+    pub rewards_per_epoch: Uint128,
+}
```

Comment:
Can we just return the `RewardsPool` object? We should also return the watermark. And for the params, can we just return the `Params` object?

Reply:
I was wondering if we should really expose our inner structs since any future change could potentially create a breaking change without us knowing. What do you think?

Reply:
I generally agree that DTOs should be disconnected from internal types most of the time. In this case the query response aggregates information from many different types, so we need a separate type anyway. I like the addition of the watermark (give it a different name so people understand what it means when they query it). I don't like the `RewardsPoolResponse` name though. I would prefer just `RewardsPool` defined here, and I'm also open to better suggestions

Reply:
that's a good point, internal structs might change, or contain information the user doesn't need/want. We should return the watermark though. It can just be called `last_distributed` or some such.

As to the name, either `RewardsPool` or `RewardsPoolInfo` is fine with me. Or just `PoolInfo`